src/iterable/dataloader/mod.rs

```diff
@@ -95,6 +95,25 @@ where
         }
         None
     }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let (lower, upper) = self.dataset_iter.size_hint();
+        if self.drop_last {
+            let lower = lower / self.batch_size;
+            let upper = upper.map(|upper| upper / self.batch_size);
+            (lower, upper)
+        } else {
+            let lower = (lower + self.batch_size - 1) / self.batch_size;
+            let upper = upper.map(|upper| (upper + self.batch_size - 1) / self.batch_size);
+            (lower, upper)
```

Comment:
Here also we can go `let upper = Some(lower)`, see the comment for `BatchSampler` for motivation.